File: src/main/java/com/itsci/mju/maebanjumpen/service/MemberService.java
package com.itsci.mju.maebanjumpen.service;

import com.itsci.mju.maebanjumpen.dto.MemberDTO;
import com.itsci.mju.maebanjumpen.model.Member;
import java.util.List;
import java.util.Optional; // เพิ่ม Optional

public interface MemberService {
    MemberDTO saveMember(MemberDTO memberDto);
    Optional<MemberDTO> getMemberById(int id); // เปลี่ยนเป็น Optional
    List<MemberDTO> getAllMembers();
    MemberDTO updateMember(int id, MemberDTO memberDto);
    void deleteMember(int id);
    MemberDTO deductBalance(int memberId, double amount);


}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/partyrole/service/MemberService.kt
package com.itsci.mju.maebanjumpen.partyrole.service

import com.itsci.mju.maebanjumpen.partyrole.dto.MemberDTO
import java.util.Optional

interface MemberService {
    fun saveMember(memberDto: MemberDTO): MemberDTO
    fun getMemberById(id: Int): Optional<MemberDTO>
    fun getAllMembers(): List<MemberDTO>
    fun updateMember(id: Int, memberDto: MemberDTO): MemberDTO
    fun deleteMember(id: Int)
    fun deductBalance(memberId: Int, amount: Double): MemberDTO
}

